internal/pkg/vm/state/cached_tree.go

```diff
@@ -60,6 +60,19 @@ func (t *CachedTree) GetOrCreateActor(ctx context.Context,
 	return actor, mappedAddr, nil
 }
 
+// DeleteActor remove the actor from the storage.
+// This method will NOT return an error if the actor was not found.
```

Comment:
It almost certainly signals a bug if it's not found.

Reply:
No, and that is not the reason that method does not error if its not found. 
Depending on the storage used, the delete operation might have a way of determining if an object exists prior to deletion. This requires doing a get first which add a cost. 
Any user wanting to know if something exists before deletion can do a get himself.

Reply:
Ok, there'll be exactly one caller in the runtime afaik, please make sure it sanity checks.